### PR TITLE
Disability care pages

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard1**
## Summary

This PR contains the schema changes, national and regional pages, and cypress tests for the Disability care feature

## Motivation

Feature request

## Detailed design

This is mainly a one-on-one copy of the verpleeghuiszorg page, so not really much else to mention here.
I've added some changes to the cypress tests based on some previous comments to tighten them up a little.

## Drawbacks

n/a

## Alternatives

n/a

## Unresolved questions

n/a

### Governance

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

File: cypress/integration/landelijk/gehandicaptenzorg.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+context('Landelijk - Gehandicaptenzorg', () => {
+  swallowResizeObserverError();
+
+  before(() => {
+    beforeNationalTests('gehandicaptenzorg');
+  });
+
+  it('Should show the correct KPI values', function (this: NationalContext) {
+    const disabilityCareLastValue = this.nationalData.disability_care
```
Comment (VWSCoronaDashboard8) [inline cypress/integration/landelijk/gehandicaptenzorg.spec.ts:15]: nit; In this context I think `lastValue` is sufficient. As long as not multiple last values are used in the same block. I think it makes the code a little more readable because it's then easier to spot differences between tests that are executing effectively the same operations.

File: cypress/integration/landelijk/intensive-care-opnames.spec.ts [hunk 2]
```diff
@@ -1,27 +1,25 @@
...
   swallowResizeObserverError();
 
   before(() => {
-    cy.fixture<National>('NL.json').as('nationalData');
-    cy.visit('/landelijk/intensive-care-opnames');
+    beforeNationalTests('intensive-care-opnames');
   });
 
   it('Should show the correct KPI values', function (this: NationalContext) {
+    const icBedsLastValue = this.nationalData.intensive_care_beds_occupied
```
Comment (VWSCoronaDashboard8) [inline cypress/integration/landelijk/intensive-care-opnames.spec.ts:15]: So same comment applies to all of these

File: cypress/integration/landelijk/positief-geteste-mensen.spec.ts [hunk 3]
```diff
@@ -1,29 +1,30 @@
...
   before(() => {
-    cy.fixture<National>('NL.json').as('nationalData');
-    cy.visit('/landelijk/positief-geteste-mensen');
+    beforeNationalTests('positief-geteste-mensen');
   });
 
   it('Should show the correct KPI values', function (this: NationalContext) {
+    const infectedTotalLastValue = this.nationalData.infected_people_total
+      .last_value;
+    const ggdLastValue = this.nationalData.ggd.last_value;
```
Comment (VWSCoronaDashboard8) [inline cypress/integration/landelijk/positief-geteste-mensen.spec.ts:17]: Here it's a different story of course.

File: src/pages/landelijk/gehandicaptenzorg.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+  return (
+    <>
+      <SEOHead
+        title={infectedLocationsText.metadata.title}
+        description={infectedLocationsText.metadata.description}
+      />
+
+      <ContentHeader
+        category={siteText.nationaal_layout.headings.kwetsbare_groepen}
+        screenreaderCategory={
```
Comment (VWSCoronaDashboard8) [inline src/pages/landelijk/gehandicaptenzorg.tsx:45]: Maybe it didn't originate from this PR, but a screen reader is not a single word.

```suggestion
        screenReaderCategory={
```

File: src/pages/landelijk/gehandicaptenzorg.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+        <KpiTile
+          title={infectedLocationsText.barscale_titel}
+          metadata={{
+            date: disabilityCareData.last_value.date_of_report_unix,
+            source: infectedLocationsText.bronnen.rivm,
+          }}
+        >
+          <KpiValue
+            data-cy="newly_infected_locations"
+            absolute={disabilityCareData?.last_value.newly_infected_locations}
```
Comment (VWSCoronaDashboard8) [inline src/pages/landelijk/gehandicaptenzorg.tsx:129]: ? should be redundant I think

**Review by VWSCoronaDashboard8 — APPROVED** [review]
A lot of work! 💯 and only a few minor suggestions 👍

File: src/pages/landelijk/gehandicaptenzorg.tsx (hunk 5 above)
Comment (VWSCoronaDashboard1) [reply src/pages/landelijk/gehandicaptenzorg.tsx:129]: Indeed, I also changed it in the verpleeghuiszorg page (where this was copied from)

File: src/pages/landelijk/gehandicaptenzorg.tsx [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+          }}
+        >
+          <KpiValue
+            data-cy="deceased_daily"
+            absolute={disabilityCareData.last_value.deceased_daily}
+          />
+        </KpiTile>
+      </TwoKpiSection>
+
+      {data && (
```
Comment (VWSCoronaDashboard10) [inline src/pages/landelijk/gehandicaptenzorg.tsx:199]: Why this check over here?

File: src/pages/landelijk/gehandicaptenzorg.tsx [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+} from '~/static-props/nl-data';
+
+const infectedLocationsText = siteText.gehandicaptenzorg_besmette_locaties;
+const positiveTestedPeopleText =
+  siteText.gehandicaptenzorg_positief_geteste_personen;
+const locationDeaths = siteText.gehandicaptenzorg_oversterfte;
+
+const DisabilityCare: FCWithLayout<NationalPageProps> = (props) => {
+  const { data } = props;
+  const disabilityCareData = data.disability_care;
```
Comment (VWSCoronaDashboard10) [inline src/pages/landelijk/gehandicaptenzorg.tsx:32]: This saves 2 characters, might be prettier to go for something like
```tsx
const values = data.disability_care.last_value
const lastValue = data.disability_care.last_value

// or

const { values, last_value } = data.disability_care

File: src/components/choropleth/tooltips/region/create-disability-infected-locations-regional-tooltip.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+) => (context: SafetyRegionProperties & RegionsDisabilityCare): ReactNode => {
+  const handler = createSelectRegionHandler(router);
+
+  const onSelect = (event: any) => {
+    event.stopPropagation();
+    handler(context);
+  };
+
+  return (
+    context && (
```
Comment (VWSCoronaDashboard10) [inline src/components/choropleth/tooltips/region/create-disability-infected-locations-regional-tooltip.tsx:20]: It looks like context is never undefined

File: src/components/choropleth/tooltips/region/create-disability-infected-locations-regional-tooltip.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+import { RegionsDisabilityCare } from '~/types/data';
+import { formatNumber, formatPercentage } from '~/utils/formatNumber';
+import { createSelectRegionHandler } from '../../select-handlers/create-select-region-handler';
+import { SafetyRegionProperties } from '../../shared';
+import { TooltipContent } from '../tooltipContent';
+
+export const createDisablityInfectedLocationsRegionalTooltip = (
+  router: NextRouter
+) => (context: SafetyRegionProperties & RegionsDisabilityCare): ReactNode => {
+  const handler = createSelectRegionHandler(router);
```
Comment (VWSCoronaDashboard10) [inline src/components/choropleth/tooltips/region/create-disability-infected-locations-regional-tooltip.tsx:12]: did you forget the second argument 'gehandicaptenzorg' over here?
Additionally, I think it would be cleaner when we don't create separate components for tooltips, instead we could render it inline, e.g.:
```diff
// src/pages/landelijk/gehandicaptenzorg.tsx

+  const selectHandler = createSelectRegionHandler(router, 'gehandicaptenzorg')
   
  // ...

  <SafetyRegionChoropleth
    metricName="disability_care"
    metricProperty="infected_locations_percentage"
-   onSelect={createSelectRegionHandler(router, 'gehandicaptenzorg')}
+   onSelect={selectHandler}
-   tooltipContent={createDisablityInfectedLocationsRegionalTooltip(
-     router
-   )}
+   tooltipContent={(context) => (
+     <TooltipContent title={context.vrname} onSelect={selectHandler}>
+       <strong>
+         {`${formatPercentage(
+           context.infected_locations_percentage
+         )}% (${formatNumber(context.infected_locations_total)})`}
+       </strong>
+     </TooltipContent>
+   )}
  />
```

but I think that can be part of a different PR

File: schema/regions/__index.json [hunk 10]
```diff
@@ -68,6 +69,13 @@
         "$ref": "nursing_home.json"
       }
     },
+    "disability_care": {
+      "type": "array",
+      "maxItems": 25,
```
Comment (VWSCoronaDashboard10) [inline schema/regions/__index.json:74]: maybe add a `minItems`?

**Review by VWSCoronaDashboard10 — APPROVED** [review]
Couple of comments/nitpicks  :-)

File: src/components/choropleth/tooltips/region/create-disability-infected-locations-regional-tooltip.tsx (hunk 9 above)
Comment (VWSCoronaDashboard10) [reply src/components/choropleth/tooltips/region/create-disability-infected-locations-regional-tooltip.tsx:12]: Ah I approved the PR, but I do think the createSelectRegionHandler in the tooltip component is missing its second argument, so that should be fixed before hitting the merge button :-)

File: src/pages/landelijk/gehandicaptenzorg.tsx (hunk 6 above)
Comment (VWSCoronaDashboard1) [reply src/pages/landelijk/gehandicaptenzorg.tsx:199]: this page is a copy/paste of verpleeghuiszorg. This check is indeed obsolete, I removed it from the verpleeghuiszorg page as well.

File: src/pages/veiligheidsregio/[code]/gehandicaptenzorg.tsx [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+          }}
+        >
+          <KpiValue
+            data-cy="deceased_daily"
+            absolute={lastValue.deceased_daily}
+          />
+        </KpiTile>
+      </TwoKpiSection>
+
+      {data && (
```
Comment (VWSCoronaDashboard4) [inline src/pages/veiligheidsregio/[code]/gehandicaptenzorg.tsx:186]: This check is redundant, data cannot be undefined; (or the uses of data elsewhere in this file can break)

**Review by VWSCoronaDashboard4 — APPROVED** [review]
